File: problems/ABC/ABC296/D.py
N, M = map(int,input().split())
INF = 10 ** 18
cnt = 0
ans = INF
for a in range(1, INF):
    b = (M + cnt) // a
    if a > b:
        break
    if b <= N and a * b >= M:
        ans = min(ans, a * b)
    cnt += 1
print(ans if ans < INF else -1)

File: problems/ABC/ABC296/D.cpp
#include <bits/stdc++.h>
using namespace std;
long long INF = powl(10, 18);
int main(){
    long long N, M;
    cin >> N >> M;
    long long ans = INF;
    int cnt = 0;
    long long b;
    for (long long a = 1; a < INF; a++) {
        b = (M + cnt) / a;
        if (a > b) break;
        if ((b <= N) && (a * b >= M)) ans = min(ans, a * b);
        cnt++;
    }
    if (ans == INF) ans = -1;
    cout << ans << endl;
}